src/hazelcore/Action_Pat.re

```diff
@@ -1560,25 +1584,39 @@ and ana_perform_operand =
         zoperand,
         ty,
       )
-    | Succeeded(zann) =>
-      let ty' = UHTyp.expand(ZTyp.erase(zann));
-      let (new_op, ctx, id_gen) =
-        Statics_Pat.ana_fix_holes_operand(ctx, id_gen, op, ty');
-      let new_zopseq = ZOpSeq.wrap(ZPat.TypeAnnZA(err, new_op, zann));
-      if (HTyp.consistent(ty, ty')) {
-        Succeeded((new_zopseq, ctx, id_gen));
-      } else {
-        let (new_zopseq, id_gen) =
-          new_zopseq |> ZPat.mk_inconsistent(id_gen);
-        Succeeded((new_zopseq, ctx, id_gen));
-      };
+    | Succeeded((zann, ctx, id_gen)) =>
+      switch (Elaborator_Typ.syn_elab(ctx, Delta.empty, ZTyp.erase(zann))) {
+      | None => Failed
+      | Some((ty', _, _)) =>
+        switch (UHPat.get_var_name(op)) {
```

Comment:
don't understand what's going on here? why are we special casing when the annotatin is on a variable?